### ISSUE TITLE
Build time too long

### ISSUE THREAD

**Issue by samsalisbury** [opentable/sous#59]
Build time is currently above 30s, which is huge for a Go project of this size. I think the main culprit is Docker and all of its dependencies. Go itself is also slower at building than it used to be (see http://dave.cheney.net/2016/04/02/go-1-7-toolchain-improvements) but I still think we can do better with Sous. I'd like to figure out how we can shorten the build time, possibly by removing or thinning out dependencies. It's also possible to get better build times by using `go install` which maintains caches and tries to do a minimal build, but I would like to reduce fresh build time if possible.

**Comment by todd** [comment]
Outsider looking in, but if you suspect that dependencies are (one of) the culprits, using Travis' container infrastructure may yield some performance gains. Unfortunately, it looks like you guys are using `sudo: required` - I assume this is so you can access the Docker daemon within Travis?

**Comment by todd** [comment]
Unless you're talking about build time locally, in which case, just ignore me 😉

**Comment by samsalisbury** [comment]
I was mainly talking about local build times... One thing I love about Go is the immediate, often <1s build times that give me constant feedback as I code, I usually compile on every save, and I'm finding the slow builds are making me write more silly mistakes than usual, as I'm building far less frequently.

I think I've found the culprit, it's `github.com/mattn/go-sqlite3` ... which by itself accounts for nearly all the build time (consistently ~31s) @nyarly, is there any other SQLite lib we could use for now?

I noticed from the `sous/...` packages, only `lib` seems to reference `go-sqlite3`. If we could at least move the dependency to somewhere in `ext` (`ext/sqlite`?) we could probably avoid building it quite so often. 

Even better may be would be to make the name cache a separate standalone app, possibly used by Sous if present, and ignored otherwise. As long as identifying images doesn't require docker metadata (i.e. labels) then the name cache shouldn't be required. (And now the `record` command is being changed to apply a Sous-friendly image name to the docker image, the metadata shouldn't be required :)

I can imagine the Docker name cache coming in very useful in general though... Docker registry badly needs search!
